libs/vscode/mcp/src/lib/init-mcp.ts

```diff
@@ -1,88 +1,98 @@
+  ExtensionContext,
+  lm,
+  version,
+  window,
+} from 'vscode';
+import type { McpStreamableWebServer } from './mcp-vscode-server';
+import type { McpCursorServer } from './mcp-cursor-server';
 import { findAvailablePort } from './ports';
 import {
   getNxWorkspacePath,
   WorkspaceConfigurationStore,
 } from '@nx-console/vscode-configuration';
-import { execSync } from 'child_process';
 import { AgentRulesManager } from './agent-rules-manager';
 import { checkIsNxWorkspace } from '@nx-console/shared-npm';
-import { vscodeLogger } from '@nx-console/vscode-output-channels';
+import {
+  logAndShowError,
+  vscodeLogger,
+} from '@nx-console/vscode-output-channels';
+import { execSync } from 'child_process';
+import {
+  getMcpJsonPath,
+  hasNxMcpEntry,
+  readMcpJson,
+  writeMcpJson,
+} from '@nx-console/vscode-utils';
 
 let mcpStreamableWebServer: McpStreamableWebServer | undefined;
+let mcpCursorServer: McpCursorServer | undefined;
 let initialized = false;
 
-export function startMcpServerSkeleton() {
-  if (shouldUseLegacyMcpRegistration()) {
-    startMcpServerSkeletonLegacy();
-    return;
-  }
-}
-
 export function stopMcpServer() {
-  if (shouldUseLegacyMcpRegistration()) {
-    stopMcpServerLegacy();
-    return;
-  }
   mcpStreamableWebServer?.stopMcpServer();
+  mcpCursorServer?.stopMcpServer();
 }
 
 export async function updateMcpServerWorkspacePath(workspacePath: string) {
-  if (shouldUseLegacyMcpRegistration()) {
-    await updateMcpServerWorkspacePathLegacy(workspacePath);
-    return;
-  }
   await mcpStreamableWebServer?.updateMcpServerWorkspacePath(workspacePath);
+  await mcpCursorServer?.updateMcpServerWorkspacePath(workspacePath);
 }
 
 export async function initMcp(context: ExtensionContext) {
   commands.executeCommand('setContext', 'isInCursor', isInCursor());
   commands.executeCommand('setContext', 'isInWindsurf', isInWindsurf());
   commands.executeCommand('setContext', 'isInVSCode', isInVSCode());
-  commands.executeCommand(
-    'setContext',
-    'isLegacyMcp',
-    shouldUseLegacyMcpRegistration(),
-  );
 
-  if (shouldUseLegacyMcpRegistration()) {
-    initMcpLegacy(context);
+  if (initialized) {
     return;
   }
+  initialized = true;
 
-  if (initialized) {
+  const inVSCode = isInVSCode();
+  const inCursor = isInCursor();
+
+  if (!inVSCode && !inCursor) {
+    vscodeLogger.log(
+      'Automatic MCP setup is only available for VSCode or Cursor.',
+    );
     return;
   }
-  initialized = true;
+  if (inVSCode && !gte(version, '1.101.0')) {
+    logAndShowError(
+      'Nx Console does not support registering the Nx MCP natively on VSCode versions older than 1.101.0. Please update VSCode',
+      'Nx Console does not support registering the Nx MCP natively on VSCode versions older than 1.101.0. Please update VSCode. We have to mark Nx Console to be compatible with older versions (even though it is not) because Cursor is based on an older version of VSCode. That is why you are seeing this error.',
```

Comment:
This is an interesting comment.. I suppose it makes sense, but calling users out like this seems aggressive?

Reply:
:O wasn't mean to be aggressive at all haha
You mean it seems accusatory for them being on an older version?